### PR TITLE
macOS install script: Support $MXNET_HOME; Print MXNet version; Added few package dependencies

### PR DESCRIPTION
**Pull request by bhavinthaker**
Changes to the installation script:
1) The macOS installation script now supports defining $MXNET_HOME to any directory of choice though the default is ~/mxnet if $MXNET_HOME is not defined.
2) Added a few package dependencies and improved the code structure.
3) Added a line to print the MXNet version.

Tested it on OSX El Capitan (Version 10.11.6) and Sierra. 

@sandeep-krishnamurthy @lxn2 @nswamy

### CODE REVIEW

File: setup-utils/install-mxnet-osx-python.sh [hunk 1]
```diff
@@ -8,32 +8,54 @@
...
+	export MXNET_TAG="v0.10.0"
+fi
+
 export TARIKH=`/bin/date +%Y-%m-%d-%H:%M:%S`
-export MXNET_HOME="$HOME/mxnet"
+if [ x"${MXNET_HOME}" = "x" ];
+then
+	export MXNET_HOME="$HOME/mxnet"
+fi
 export MXNET_HOME_OLD="$HOME/mxnet_${TARIKH}"
```
Comment (sandeep-krishnamurthy) [inline setup-utils/install-mxnet-osx-python.sh]: If new mxnet install folder is provided, should we create "old" directory there?

File: setup-utils/install-mxnet-osx-python.sh [hunk 2]
```diff
@@ -8,32 +8,54 @@
...
 LINE="########################################################################"
 
 echo $LINE
 echo " "
-echo "This script installs MXNet on MacOS in ${MXNET_HOME}"
-echo "If this directory is already present, it is renamed to ${MXNET_HOME_OLD}"
-echo "It has been tested to work successfully on MacOS El Capitan and Sierra"
-echo "and is expected to work fine on other versions as well."
+echo "This script installs MXNet on MacOS in \${MXNET_HOME}"
+echo "The default value of \${MXNET_HOME} = ~/mxnet but you can set it appropriately."
```
Comment (sandeep-krishnamurthy) [inline setup-utils/install-mxnet-osx-python.sh]: May be - "You can set it with <this environment variable>?

File: setup-utils/install-mxnet-osx-python.sh [hunk 3]
```diff
@@ -62,34 +84,83 @@ runme() {
...
 runme brew_pkg_install pkg-config
 runme brew_pkg_install python
-runme pip install --upgrade pip
-brew install homebrew/science/openblas
 runme brew_pkg_install opencv
-# Needed for /usr/local/lib/graphviz to be created
-runme brew_pkg_install graphviz
 runme brew_pkg_install numpy
+runme brew_pkg_install homebrew/science/openblas
+#brew install homebrew/science/openblas
```
Comment (sandeep-krishnamurthy) [inline setup-utils/install-mxnet-osx-python.sh]: Can we remove commented code snippets?

File: setup-utils/install-mxnet-osx-python.sh [hunk 4]
```diff
@@ -8,32 +8,54 @@
 
 #set -ex
 
+export MXNET_GITPATH="https://github.com/dmlc/mxnet.git"
+if [ x"${MXNET_TAG}" = "x" ];
```
Comment (chunyang-wen) [inline setup-utils/install-mxnet-osx-python.sh]: [ -z ${variable_name} ]

Comment (bhavinthaker) [inline setup-utils/install-mxnet-osx-python.sh]: Agreed, though this change should be fine too, correct?

File: setup-utils/install-mxnet-osx-python.sh (hunk 1 above)
Comment (bhavinthaker) [inline setup-utils/install-mxnet-osx-python.sh]: An old directory does get created by the lines: 
if [ -d ${MXNET_HOME} ]; then
                mv ${MXNET_HOME} ${MXNET_HOME_OLD}

File: setup-utils/install-mxnet-osx-python.sh (hunk 3 above)
Comment (bhavinthaker) [inline setup-utils/install-mxnet-osx-python.sh]: Good point. I will do so.

File: setup-utils/install-mxnet-osx-python.sh (hunk 2 above)
Comment (bhavinthaker) [inline setup-utils/install-mxnet-osx-python.sh]: Ok, I changed the comment to "If not, the default value is ..."

**Pull request merged**